src/scanner/mod.rs

```diff
@@ -36,13 +40,20 @@ impl Scanner {
         ips: &[IpAddr],
         batch_size: u16,
         timeout: Duration,
+        tries: u8,
         greppable: bool,
         port_strategy: PortStrategy,
         accessible: bool,
     ) -> Self {
         Self {
             batch_size,
             timeout,
+            tries: if tries == 0 {
+                eprintln!("tries is 0, corrected to 1!");
+                tries + 1
```

Comment:
Why not just `1`? By the way, if you disallow zero here, I think [`NonZeroU8`](https://doc.rust-lang.org/std/num/struct.NonZeroU8.html) would make a great API and it would remove that last part in the `scan_socket` function. I would still keep the constructor argument as `u8` though and correct in here like you do.

Reply:
I corrected it on my side.
About `NonZeroU8` why not, but I prefer correcting it for the user instead of exploding because it is impossible to set a `NonZeroU8` to 0.

Reply:
No, I meant if you were to make `self.tries` a `NonZeroU8`, you could have that guarantee in the `scan_socket` method. And of course, you would still correct the 0 here

Reply:
I switch to `NonZeroU8` for `tries` inside `Scanner` on other locations (*e.g.* `input.rs`) `tries` is still an `u8`.